src/vs/workbench/parts/preferences/electron-browser/settingsEditor2.ts

```diff
@@ -47,6 +47,7 @@ import { CONTEXT_SETTINGS_EDITOR, CONTEXT_SETTINGS_SEARCH_FOCUS, CONTEXT_TOC_ROW
 import { IPreferencesService, ISearchResult, ISettingsEditorModel, ISettingsEditorOptions, SettingsEditorOptions, SettingValueType } from 'vs/workbench/services/preferences/common/preferences';
 import { SettingsEditor2Input } from 'vs/workbench/services/preferences/common/preferencesEditorInput';
 import { Settings2EditorModel } from 'vs/workbench/services/preferences/common/preferencesModels';
+import { Disposable } from 'vscode';
```

Comment:
Don't import from 'vscode' here, that module is for extensions. You can use IDisposable from lifecycle.ts

Reply:
Actually you don't need to add that array because SettingsEditor2 already extends Disposable, the _register method adds disposable objects to its internal set of disposables.